src/adaptations/device-layer/include/Weave/DeviceLayer/internal/GenericSoftwareUpdateManagerImpl.ipp

```diff
@@ -441,20 +443,17 @@ void GenericSoftwareUpdateManagerImpl<ImplClass>::_OnPlatformEvent(const WeaveDe
 {
     GenericSoftwareUpdateManagerImpl<ImplClass> * self = &SoftwareUpdateMgrImpl();
 
-     if (event->Type == DeviceEventType::kServiceConnectivityChange)
-     {
+    if (event->Type == DeviceEventType::kServiceConnectivityChange)
```

Comment:
The ConnectivityManager provides a HaveServiceConnectivity() method.  So I think we can eliminate all the event handling logic here, along with mHaveServiceConnectivity.

Reply:
Done.  Thanks for the suggestion.   I perhaps cut a bit more than initially intended -- I eliminated all references to `GenericSoftwareUpdateManagerImpl<>::_OnPlatformEvent`.  This eliminates it from the pattern of having the PlatformManager forward calls to specific managers on platform events, but it is clear how to recreate such a pattern when the need arises again.

Reply:
Agreed.  I think its cleaner not to forward OnPlatformEvent() if it's not really needed.